src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/PermalinkService.cs

```diff
@@ -96,4 +114,42 @@ private async Task<PermalinkViewModel> BuildViewModel(Permalink permalink)
             return viewModel;
         }
     }
+
+    internal class PermalinkContractResolver : DefaultContractResolver
+    {
+        private readonly bool _tableResultLocationHierarchiesEnabled;
+
+        public PermalinkContractResolver(bool tableResultLocationHierarchiesEnabled)
+        {
+            _tableResultLocationHierarchiesEnabled = tableResultLocationHierarchiesEnabled;
+        }
+
+        protected override JsonProperty CreateProperty(MemberInfo member, MemberSerialization memberSerialization)
+        {
+            JsonProperty property = base.CreateProperty(member, memberSerialization);
+            if (property.DeclaringType == typeof(Permalink))
+            {
+                property.ShouldSerialize = property.PropertyName switch
+                {
+                    "Locations" => _ => !_tableResultLocationHierarchiesEnabled,
+                    "HierarchicalLocations" => _ => _tableResultLocationHierarchiesEnabled,
```

Comment:
Should this not be `LocationsHierarchical`?

Reply:
Good spot! Corrected now, thanks 🙂 . So far I only did some testing with the feature turned off, I will do some testing with it turned on now.

Reply:
Nice one.